replica-node/src/main.rs

```diff
@@ -0,0 +1,356 @@
+//! The main AccountsDb replication node responsible for replicating
+//! AccountsDb information from peer a validator or another replica-node.
+
+#![allow(clippy::integer_arithmetic)]
+
+use {
+    clap::{crate_description, crate_name, value_t, values_t, App, AppSettings, Arg},
+    log::*,
+    rand::{seq::SliceRandom, thread_rng},
+    solana_clap_utils::{
+        input_parsers::keypair_of,
+        input_validators::{is_keypair_or_ask_keyword, is_parsable, is_pubkey},
+        keypair::SKIP_SEED_PHRASE_VALIDATION_ARG,
+    },
+    solana_gossip::{
+        cluster_info::{Node, VALIDATOR_PORT_RANGE},
+        contact_info::ContactInfo,
+    },
+    solana_replica_node::{
+        replica_node::{ReplicaNode, ReplicaNodeConfig},
+        replica_util,
+    },
+    solana_sdk::{pubkey::Pubkey, signature::Signer},
+    solana_validator::port_range_validator,
+    std::{
+        collections::HashSet,
+        env,
+        net::{IpAddr, SocketAddr},
+        path::PathBuf,
+        process::exit,
+    },
+};
+
+pub fn main() {
+    let default_dynamic_port_range =
+        &format!("{}-{}", VALIDATOR_PORT_RANGE.0, VALIDATOR_PORT_RANGE.1);
+
+    let matches = App::new(crate_name!())
+        .about(crate_description!())
+        .version(solana_version::version!())
+        .setting(AppSettings::VersionlessSubcommands)
+        .setting(AppSettings::InferSubcommands)
+        .arg(
+            Arg::with_name(SKIP_SEED_PHRASE_VALIDATION_ARG.name)
+                .long(SKIP_SEED_PHRASE_VALIDATION_ARG.long)
+                .help(SKIP_SEED_PHRASE_VALIDATION_ARG.help),
+        )
+        .arg(
+            Arg::with_name("ledger_path")
+                .short("l")
+                .long("ledger")
+                .value_name("DIR")
+                .takes_value(true)
+                .required(true)
+                .default_value("ledger")
+                .help("Use DIR as ledger location"),
+        )
+        .arg(
+            Arg::with_name("peer")
+                .long("peer")
+                .value_name("IP:PORT")
+                .takes_value(true)
+                .required(true)
+                .help("The the IP:PORT for the peer validator/replica to download from"),
+        )
+        .arg(
+            Arg::with_name("peer_pubkey")
```

Comment:
Should this Arg be required? We use `unwrap()` on it below L176

Reply:
For now it is required.

Reply:
Great, please add `.required(true)`

Reply:
Done